### PR TITLE
Make 'cursorline' apply to 'breakindent' and 'showbreak' consistently

### PR DESCRIPTION
**Pull request by zeertzjq**
According to #3355 `'cursorline'` should not apply to `'breakindent'`, but it applies to `'breakindent'` in diff mode when `'cursorlineopt'` contains `line`. This PR makes it so `'cursorline'` never applies to `'breakindent'`.

When `'cursorlineopt'` contains `screenline`, `'cursorline'` doesn't apply to `'showbreak'` when `'number'` is on or when `'breakindent'` is on and `'breakindentopt'` doesn't contain `sbr`, but it applies to `'showbreak'` **on the next screenline** when `number` is off and some other conditions are satisfied (e.g. `'breakindent'` is off or `'breakindentopt'` contains `sbr`). This PR makes it so `'cursorline'` never applies to `'showbreak'` when `'cursorlineopt'` contains `screenline`.

### CODE REVIEW

**Comment by chrisbra** [comment]
can we have a test?

**Comment by chrisbra** [comment]
Okay, I'll review tomorrow

**Comment by zeertzjq** [comment]
> Okay, I'll review tomorrow

@chrisbra

**Comment by chrisbra** [comment]
thanks, I have to delay reviewing it. My day job is keeping me very busy currently. I'll try to squeeze it in this week.

**Comment by chrisbra** [comment]
Thanks the change to `drawline.c` looks good. I have the following suggestion for the `test_cursorline` however. Can you please squash in the following change on top of your patch:
```diff
diff --git a/src/testdir/test_cursorline.vim b/src/testdir/test_cursorline.vim
index 70da1f2c6..0e62284f3 100644
--- a/src/testdir/test_cursorline.vim
+++ b/src/testdir/test_cursorline.vim
@@ -196,6 +196,7 @@ func Test_cursorline_screenline()
   endif
   " showbreak should not be highlighted with CursorLine when 'number' is off
   call term_sendkeys(buf, "gg0")
+  call term_sendkeys(buf, ":set list cursorlineopt=screenline listchars=space:-\<cr>")
   call term_sendkeys(buf, ":set nonumber\<cr>")
   call VerifyScreenDump(buf, 'Test_'. filename. '_19', {})
   call term_sendkeys(buf, "fagj")
@@ -213,6 +214,7 @@ func Test_cursorline_screenline()
   call term_sendkeys(buf, "gj")
   call TermWait(buf)
   call VerifyScreenDump(buf, 'Test_'. filename. '_24', {})
+  call term_sendkeys(buf, ":set list& cursorlineopt& listchars&<cr>")

   call StopVimInTerminal(buf)
   call delete(filename)
```

This just makes sure, that the test starts and ends with a clear state (and resets the changes from the beginning of the test (which I forgot to undo :( )

**Review by chrisbra — APPROVED** [review]
Looks good, except for minor remark regarding test_cursorline

**Review by chrisbra — APPROVED** [review]

File: src/testdir/test_cursorline.vim [hunk 1]
```diff
@@ -192,7 +192,29 @@ func Test_cursorline_screenline()
...
+  call term_sendkeys(buf, "gj")
+  call TermWait(buf)
+  call VerifyScreenDump(buf, 'Test_'. filename. '_22', {})
+  call term_sendkeys(buf, "gj")
+  call TermWait(buf)
+  call VerifyScreenDump(buf, 'Test_'. filename. '_23', {})
+  call term_sendkeys(buf, "gj")
+  call TermWait(buf)
+  call VerifyScreenDump(buf, 'Test_'. filename. '_24', {})
+  call term_sendkeys(buf, ":set list& cursorlineopt& listchars&<cr>")
```
Comment (chrisbra) [inline src/testdir/test_cursorline.vim]: > call term_sendkeys(buf, ":set list& cursorlineopt& listchars&<cr>")

excuse me, this was probably my fault. There is a backslash missing before the `<cr>` Sorry :(

**Pull request closed without merging**